### PR TITLE
Detect stale `commitInGitGit` and force-update it

### PR DESCRIPTION
**Pull request by dscho**
I just noticed that https://github.com/gitgitgadget/git/pull/453 has not been updated to reflect that the patch was merged into `next`. Looks like a stale `commitInGitGit` for the head commit is still recorded, and therefore GitGitGadget is no longer able to service that PR correctly.

Let's try to detect such a situation by verifying that the commit in question is actually in `pu`.

### CODE REVIEW

File: lib/ci-helper.ts [hunk 1]
```diff
@@ -144,6 +144,12 @@ export class CIHelper {
             return false;
         }
 
+        const commitsInPu: Set<string> = new Set<string>(
+            (await git(["rev-list", "--no-merges",
+                        "^refs/remotes/upstream/maint~100",
```
Comment (webstech) [inline lib/ci-helper.ts]: Should `refs/heads/maint` be added to the git fetch at the start of this method?

File: lib/ci-helper.ts [hunk 2]
```diff
@@ -160,7 +166,14 @@ export class CIHelper {
             }
             const meta = await this.getMailMetadata(messageID);
             if (!meta || meta.commitInGitGit !== undefined) {
-                continue;
+                if (!meta || commitsInPu.has(meta.commitInGitGit!)) {
+                    continue;
+                }
+                console.log(`Upstream commit ${meta.commitInGitGit} for ${
+                    info.headCommit} of ${
+                    info.pullRequestURL} no found longer in pu`);
```
Comment (webstech) [inline lib/ci-helper.ts]: Should probably be `no longer found in pu`.

**Comment by webstech** [comment]
The commit message has a typo: `preformance`.  May as well correct when making updates.

File: lib/ci-helper.ts (hunk 1 above)
Comment (dscho) [reply lib/ci-helper.ts]: Good point!

File: lib/ci-helper.ts (hunk 2 above)
Comment (dscho) [reply lib/ci-helper.ts]: Oy. 🤦‍♂

**Pull request merged**